src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -1151,15 +1151,25 @@ public function originalIsEquivalent($key)
 
         if ($attribute === $original) {
             return true;
-        } elseif (is_null($attribute)) {
-            return false;
-        } elseif ($this->isDateAttribute($key)) {
-            return $this->fromDateTime($attribute) ===
-                   $this->fromDateTime($original);
         }
 
+        // If the attribute is a date, compare it by datetime format
+        if ($this->isDateAttribute($key)) {
+            return $this->fromDateTime($attribute) === $this->fromDateTime($original);
+        }
+
+        // If the original or attribute is an object, compare it by string
+        if (is_object($attribute)
+            || is_object($original)) {
+            $attributeAsString = is_object($attribute) ? json_encode($attribute) : $attribute;
+            $originalAsString = is_object($original) ? json_encode($original) : $original;
```

Comment:
I don't have a better suggestion but json encoding for comparison looks quite expensive ... is there no better way?

Reply:
Yes your right. As it seems a non strict comparison would solve the same problem.